File: jfingerprintmanager/src/main/java/com/jesusm/jfingerprintmanager/encryption/presenter/FingerprintEncryptionDialogPresenter.java
package com.jesusm.jfingerprintmanager.encryption.presenter;

import android.support.annotation.NonNull;
import android.support.v4.hardware.fingerprint.FingerprintManagerCompat;

import com.jesusm.jfingerprintmanager.base.FingerprintBaseDialogPresenter;

public class FingerprintEncryptionDialogPresenter extends FingerprintBaseDialogPresenter {
    @NonNull
    protected View view;

    public FingerprintEncryptionDialogPresenter(@NonNull View view) {
        super(view);
        this.view = view;
    }

    @Override
    public void onViewShown() {
        if (!fingerprintHardware.isFingerprintAuthAvailable()) {
            close();
            return;
        }

        super.onViewShown();
    }

    @Override
    public void onAuthenticationSucceeded(FingerprintManagerCompat.AuthenticationResult result) {
        view.onAuthenticationSucceed(result.getCryptoObject());
        close();
    }

    public interface View extends FingerprintBaseDialogPresenter.View {
        void onAuthenticationSucceed(FingerprintManagerCompat.CryptoObject cryptoObject);
    }
}

File: jfingerprintmanager/src/main/kotlin/com/jesusm/jfingerprintmanager/encryption/presenter/FingerprintEncryptionDialogPresenter.kt
package com.jesusm.jfingerprintmanager.encryption.presenter

import android.support.v4.hardware.fingerprint.FingerprintManagerCompat
import com.jesusm.jfingerprintmanager.base.ui.presenter.FingerprintBaseDialogPresenter

class FingerprintEncryptionDialogPresenter(view : View) : FingerprintBaseDialogPresenter(view) {
    interface View : FingerprintBaseDialogPresenter.View {
        fun onAuthenticationSucceed(cryptoObject: FingerprintManagerCompat.CryptoObject)
    }

    override fun onViewShown() {
        fingerprintHardware?.apply {
            if (isFingerprintAuthAvailable().not()) {
                close()
                return
            }
        }

        super.onViewShown()
    }

    override fun onAuthenticationSucceeded(result: FingerprintManagerCompat.AuthenticationResult?) {
        super.onAuthenticationSucceeded(result)

        result?.let {
            (view as View).onAuthenticationSucceed(it.cryptoObject)
        }
        close()
    }
}